include/roboteam_utils/Angle.h

```diff
@@ -12,218 +12,117 @@ namespace rtt {
-         * @return Angle this->angle + scalar
-         */
-        Angle operator+(const double &scalar) const noexcept;
-
-        /**
-         * @brief Subtracts two angles
-         *
-         * @param other Angle to subtract
-         * @return Angle this->angle - other.angle
+         * Compute the subtraction between this Angle and the other Angle, and directly rescale the angle value.
+         * @param other The other Angle.
+         * @return A new Angle instance which is the result of the subtraction.
          */
         Angle operator-(const Angle &other) const noexcept;
 
         /**
-         * @brief Subtracts two angles
-         *
-         * @param scalar Angle to subtract
-         * @return Angle this->angle - other.angle
-         */
-        Angle operator-(const double &scalar) const noexcept;
-
-        /**
-         * @brief Sets the current angle equal to`this->angle + other.angle`
-         *
-         * @param other Other angle to add to `this`
-         * @return Angle A copy of `*this`
+         * Add the other Angle to this Angle, and directly rescale the angle value.
+         * @param other The other Angle.
+         * @return A reference to this updated Angle instance.
          */
         Angle operator+=(const Angle &other) noexcept;
 
         /**
-         * @brief Sets the current angle equal to `this->angle + scalar`
-         *
-         * @param scalar Other angle to add to `this`
-         * @return Angle A copy of `*this`
-         */
-        Angle operator+=(const double &scalar) noexcept;
-
-        /**
-         * @brief Sets the current angle equal to `this->angle - other.angle`
-         *
-         * @param other Other angle to subtract from `this`
-         * @return Angle A copy of `*this`
+         * Subtract the other Angle from this Angle, and directly rescale the angle value.
+         * @param other The other Angle.
+         * @return A reference to this updated Angle instance.
          */
         Angle operator-=(const Angle &other) noexcept;
 
         /**
-         * @brief Sets the current angle equal to `this->angle - scalar`
-         *
-         * @param scalar Other angle to subtract from `this`
-         * @return Angle A copy of `*this`
-         */
-        Angle operator-=(const double &scalar) noexcept;
-
-        /**
-         * @brief Copy assignment operator
-         *
-         * Explicitly constraints.
-         *
-         * @param scalar Scalar to set `this->angle` to
+         * Set the angle value of this Angle instance, which will be directly rescaled.
+         * @param scalar A double value which represent the new angle value. It is allowed to use a value outside the range [-PI, PI). However you should be cautious with float
+         * values close to -MAX_FLOAT and close to MAX_FLOAT.
+         * @return A reference to this updated Angle instance.
          */
         Angle &operator=(const double &scalar) noexcept;
```

Comment:
This is quite inconsistent with the behavior of the other -= operator which returns an instance and not a reference

Reply:
Also, why is there no `Angle &operator=(const Angle &angle);`?

Reply:
Well, since it is not used...